### ISSUE TITLE
bug: signature_help error 

### ISSUE THREAD

**Issue by Fau818** [folke/noice.nvim#435]
### Did you check docs and existing issues?

- [X] I have read all the noice.nvim docs
- [X] I have searched the existing issues of noice.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.9.0 Release

### Operating system/version

MacOS 13.3.1

### Describe the bug

In some cases, the latest version of Noice will give an error when showing the signature help.

The error message is shown below.

```
...l/share/nvim/lazy/noice.nvim/lua/noice/lsp/signature.lua:221: attempt to concatenate field 'label' (a table value)
```

Here is a video to show this issue. And there is no problem with this commit at `1da4edf`.

https://user-images.githubusercontent.com/75115028/232697234-d6a3460b-0f6a-4bad-905a-ff28733c691c.mov

### Steps To Reproduce

...

### Expected Behavior

...

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/noice.nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by folke** [comment]
Should be fixed now. Would be great if you could confirm. Thanks!

**Comment by Fau818** [comment]
@folke Unfortunately, I just tested it, and now the error is on line 223.

**Comment by folke** [comment]
Can you give me some example code?

**Comment by Fau818** [comment]
> Can you give me some example code?

It seems that the error occurs because the `label` field is a **table**.

Here is an example in python.

```python
import torch

torch.hub.load()
```

**Comment by folke** [comment]
test again

**Comment by Fau818** [comment]
> test again

Okay, it's working great now!
